Mathlib/RingTheory/Ideal/Basic.lean

```diff
@@ -641,6 +641,36 @@ theorem span_pow_eq_top (s : Set α) (hs : span s = ⊤) (n : ℕ) :
   exact ⟨f x ^ (n + 1), mul_comm _ _⟩
 #align ideal.span_pow_eq_top Ideal.span_pow_eq_top
 
+lemma isPrime_of_maximally_disjoint (I : Ideal α)
+    (S : Submonoid α) (disjoint : (I : Set α) ⊓ S = ∅)
+    (maximally_disjoint : ∀ (J : Ideal α), I < J → (J : Set α) ⊓ S ≠ ∅) :
```

Comment:
@Yael, do we have some predicate that combines these two, as the more general
```lean
    (p : P I)
    (maximally_p : ∀ (J : Ideal α), P J → I < J → False) :
```

Reply:
There's `IsMaxOn`, but weirdly it uses `Filter` 🤔

Reply:
I tried using `IsMaxOn` here but the API is very much lacking. For example, you cannot easily get Eric's `maximally_p` out of it. So I think the current spelling of `maximally_disjoint` is okay, if we switch to `Disjoint`:
```suggestion
    (S : Submonoid α) (disjoint : Disjoint (I : Set α) S)
    (maximally_disjoint : ∀ (J : Ideal α), I < J → ¬ Disjoint (J : Set α) S) :
```